cura/Settings/GlobalStack.py

```diff
@@ -131,6 +131,14 @@ def getConfigurationTypeFromSerialized(cls, serialized: str) -> Optional[str]:
             return "machine_stack"
         return configuration_type
 
+    def getIntentCategory(self) -> str:
+        intent_category = "default"
+        for extruder in self.extruderList:
+            category = extruder.intent.getMetaDataEntry("intent_category", "default")
+            if category != "default" and category != intent_category:
+                intent_category = category
+        return intent_category
```

Comment:
This code duplicates `MachineManager.activeIntentCategory`. We need to remove either one.

Reply:
`MachineManager.activeIntentCategory` calls this function to get the intent cateogry. It also checks if the active global stack is present and always returns a string.